functable.c

```diff
@@ -213,6 +216,10 @@ ZLIB_INTERNAL uint32_t adler32_stub(uint32_t adler, const unsigned char *buf, si
     if (x86_cpu_has_avx2)
         functable.adler32 = &adler32_avx2;
 #endif
+#ifdef POWER8
+    if (power_cpu_has_arch_2_07)
+        functable.adler32 = &adler32_power8;
```

Comment:
Maybe the function should be `adler32_vsx`?  Named after the instruction set rather than the cpu..

Reply:
We choose to use `adler32_power8` because we intend to not support old machines < P8. However, this code may works on P7 (not sure). Should work on P9.

Reply:
We initially opted for the `*_powerX` naming scheme because that better reflects a set of features on POWER. The set of VSX instructions is usually incremented with each new version of the Power ISA, so the best way to distinguish an specific feature/instruction set is by the ISA version, which usually maps to a processor version (e.g. P7, P8, P9). `adler32_power8` indicates that P8 (ISA 2.07) is the baseline, although it also runs on later processor versions, as the instruction sets are incremental. So I believe that keeping the `powerX` suffix would be best.

Reply:
Ok sounds good.